gateway-contracts/tasks/pauseContracts.ts

```diff
@@ -0,0 +1,172 @@
+import dotenv from "dotenv";
+import { Wallet } from "ethers";
+import fs from "fs";
+import { task, types } from "hardhat/config";
+import { HardhatEthersHelpers } from "hardhat/types";
+import path from "path";
+
+import { ADDRESSES_DIR } from "../hardhat.config";
+import { getRequiredEnvVar } from "./utils/loadVariables";
+import { pascalCaseToSnakeCase } from "./utils/stringOps";
+
+// Helper function to pause a contract
+async function pauseContract(name: string, ethers: HardhatEthersHelpers, useInternalAddress: boolean) {
+  // Get the pauser wallet
+  const pauserPrivateKey = getRequiredEnvVar("PAUSER_PRIVATE_KEY");
+  const pauser = new Wallet(pauserPrivateKey).connect(ethers.provider);
+
+  // Get contract factories
+  if (useInternalAddress) {
+    const envFilePath = path.join(ADDRESSES_DIR, `.env.gateway`);
+
+    if (!fs.existsSync(envFilePath)) {
+      throw new Error(`Environment file not found: ${envFilePath}`);
+    }
+
+    dotenv.config({ path: envFilePath, override: true });
+  }
+
+  // Determine env variable name for the proxy contract address
+  const nameSnakeCase = pascalCaseToSnakeCase(name);
+  const addressEnvVarName = `${nameSnakeCase.toUpperCase()}_ADDRESS`;
+
+  // Get the proxy address
+  const proxyAddress = getRequiredEnvVar(addressEnvVarName);
+  if (!proxyAddress) {
```

Comment:
@RomanBredehoft, I just realized that these validations (`!proxyAddress`) are not necessary since the `getRequiredEnvVar` already throws an error if the value is not present in `process.env`